tasks/ast_tools/src/generators/typescript.rs

```diff
@@ -27,20 +27,30 @@ impl Generator for TypescriptGenerator {
         let estree_derive_id = codegen.get_derive_id_by_name("ESTree");
 
         let mut code = String::new();
+        let mut ast_node_names: Vec<String> = vec![];
         for type_def in &schema.types {
             if type_def.generates_derive(estree_derive_id) {
-                generate_ts_type_def(type_def, &mut code, schema);
+                generate_ts_type_def(type_def, &mut code, schema, &mut ast_node_names);
             }
         }
 
+        // manually append `FormalParameterRest`, which is generated via `add_ts_def`.
+        let ast_node_union = ast_node_names.join(" | ");
+        write_it!(code, "export type Node = {ast_node_union} | FormalParameterRest;\n\n");
```

Comment:
Not ideal to hard-code addition of `FormalParameterRest` here. We *could* parse `add_ts_type` string and get the type name from it.

But `add_ts_def` is a hack anyway, and we should find something better to replace it. So not worthwhile to implement `add_ts_type` parsing now.

Please just add a `// TODO: Should not be hard-coded here.` comment for now.